src/shared/schema/temp_lead_registration.schema.ts

```diff
@@ -0,0 +1,22 @@
+import { Prop, Schema, SchemaFactory } from "@nestjs/mongoose";
+import { Document } from "mongoose";
+import { UserRole } from "../interfaces";
+
+@Schema()
+export class TempLeadRegistration extends Document { 
+  @Prop ({required: true})
+  email: string;
+
+  @Prop ({required: true})
+  role:string = UserRole.LEAD;
+
+  @Prop ({required: true})
+  CreatedAt: Date;
+
+  @Prop ({required: true})
+  userId: string;
+
+  @Prop ({required: true})
+  leadPosition: string;
```

Comment:
I'm asumming temp-lead is for those that are requesting to join the community as a lead,
Include their first and last name here, and all other details, 

1. Firstname 
2. Lastname 
3. Email.
4. whyJoinUs
5. Location.
6. Student or Porfessional 
7. Links to online profiles.
8. Cv file upload

I think this details sholud be enough for anyone planning to join us.

Reply:
Temp-lead holds tempory user(leads) applications.
the pending applications that an admin can view are all stored here.
When an applicationis approved, a new lead is created  and the bulk of its data is gotten form the users or the provided form -for new users-